### PR TITLE
feat: `v2` Swap API

### PR DESCRIPTION
**Pull request by 0xAlec**
**What changed? Why?**
- add `aggregators` prop to `<Swap/>` to control `V1` or `V2` API
- this will add the `v2Enabled` parameter in the request body, our API will route these to a beta implementation for serving quotes/trades directly from Uniswap V3

`V1` API does not index pools with under <= $10,000 of liquidity, `V2` API will make a direct call to the `UniswapRouter` contract

**Notes to reviewers**
`slippage` for the `V2` API is hardcoded to 1000 bps (10%) - Tina is aligned for the initial launch and we'll fast-follow to add configurability to the UI

usage
```javascript
<Swap address={address} experimental={{ useAggregator: false }}>
          ...
</Swap>
```

**How has it been tested?**
See Slack for demo

### CODE REVIEW

File: src/swap/types.ts [hunk 1]
```diff
@@ -57,6 +58,7 @@ export type GetSwapAPIParams = GetQuoteAPIParams & {
 export type GetSwapQuoteParams = {
   from: Token; // The source token for the swap
   to: Token; // The destination token for the swap
+  aggregator: boolean; // Whether to use a DEX aggregator
```
Comment (Zizzamia) [inline src/swap/types.ts:61]: alphabetical

File: src/swap/types.ts [hunk 2]
```diff
@@ -45,6 +45,7 @@ export type GetQuoteAPIParams = {
   to: AddressOrETH | ''; // The destination address or 'ETH' for Ethereum
   amount: string; // The amount to be swapped
   amountReference?: string; // The reference amount for the swap
+  v2Enabled?: boolean; // Whether to use V2 of the API (default: false)
```
Comment (Zizzamia) [inline src/swap/types.ts:48]: why not just enable it as default? After all it should not have a real change for the experience, right?

File: src/swap/utils/buildSwapTransaction.ts [hunk 3]
```diff
@@ -23,7 +23,14 @@ export async function buildSwapTransaction(
     isAmountInDecimals: false,
   };
 
-  const apiParams = getAPIParamsForToken({ ...defaultParams, ...params });
+  let apiParams = getAPIParamsForToken({ ...defaultParams, ...params });
+
+  if (!params.aggregator) {
+    apiParams = {
+      v2Enabled: true,
```
Comment (Zizzamia) [inline src/swap/utils/buildSwapTransaction.ts:30]: oh I see you want to test this on his own for a bit?

File: src/swap/types.ts (hunk 1 above)
Comment (0xAlec) [reply src/swap/types.ts:59]: fixed

File: src/swap/types.ts (hunk 2 above)
Comment (0xAlec) [reply src/swap/types.ts:48]: see Slack

File: src/swap/utils/buildSwapTransaction.ts (hunk 3 above)
Comment (0xAlec) [reply src/swap/utils/buildSwapTransaction.ts:30]: yeah, we'll do a switchover once the V2 service is more production ready

File: src/swap/components/SwapProvider.tsx [hunk 4]
```diff
@@ -36,9 +36,11 @@ export function useSwapContext() {
 export function SwapProvider({
   address,
   children,
+  useAggregator,
```
Comment (Zizzamia) [inline src/swap/components/SwapProvider.tsx:39]: the provider too should have `experimental`

Comment (Zizzamia) [reply src/swap/components/SwapProvider.tsx:39]: we don't want to leak bad experience across too many components, and we want to keep `experimental` nice and neat.

File: src/swap/types.ts (hunk 2 above)
Comment (Zizzamia) [reply src/swap/types.ts:48]: use `experimental`

File: src/swap/types.ts [hunk 5]
```diff
@@ -55,11 +56,12 @@ export type GetSwapAPIParams = GetQuoteAPIParams & {
...
  */
 export type GetSwapQuoteParams = {
-  from: Token; // The source token for the swap
-  to: Token; // The destination token for the swap
   amount: string; // The amount to be swapped
   amountReference?: string; // The reference amount for the swap
+  from: Token; // The source token for the swap
   isAmountInDecimals?: boolean; // Whether the amount is in decimals
+  to: Token; // The destination token for the swap
+  useAggregator: boolean; // Whether to use a DEX aggregator
```
Comment (Zizzamia) [inline src/swap/types.ts:64]: use experimental

File: src/swap/components/SwapProvider.tsx [hunk 6]
```diff
@@ -36,10 +36,17 @@ export function useSwapContext() {
...
+  experimental,
 }: {
-  children: React.ReactNode;
   address: Address;
+  children: React.ReactNode;
+  experimental: {
+    useAggregator: boolean; // Whether to use a DEX aggregator. (default: true)
+  };
 }) {
+  // Feature flags
```
Comment (Zizzamia) [inline src/swap/components/SwapProvider.tsx:47]: Love the comment

File: src/swap/types.ts (hunk 2 above)
Comment (Zizzamia) [reply src/swap/types.ts:48]: `v2Enabled` this is such a bad way to name a property for an API.

Comment (Zizzamia) [reply src/swap/types.ts:48]: do we really need this year?

**Review by Zizzamia — APPROVED** [review]

**Pull request merged**